Various small fixes/tweaks

classicrocker883 opened the pull request:
<!--

Submitting a Pull Request

- Please fill out all sections of this form. You can delete the helpful comments.
- Pull Requests without clear information will take longer and may even be rejected.
- We get a high volume of submissions so please be patient during review.

-->

### Description
Most file changes remove `HAS_STATUS_MESSAGE` swap w/`HAS_DISPLAY`
Also, 2nd most are changes to whitespace, and comments.

- remove whitespace/fix wording, tweak spacing  
- ~~add hyphen (-) to "Gcode" -- (follow up to "📝 GCode => G-Code")~~ i believe done in another **PR**  
- ~~remove unused `INLINE_USART_IRQ`~~  done in another **PR**  
- optimize code - add `MMS_SCALED` to `feedrate_mm_s` + simplified the PID Plot code in **proui/dwin.cpp** so it's easier to read
- add a Default-No Probe/No Mesh test to **buildroot/tests/STM32F103RE_creality**
- Removed `COLOR_AQUA` because it is unused, and also the same as `COLOR_CYAN`
- add `HAS_TRAMMING_WIZARD`
- ~~`set_status_and_level(..., level=0)` from #26181~~ done in another **PR** 
<br>

- - this should be looked at if this change is needed vs change in other **PR**(see *Note*)
- - same thing change for **M48.cpp**
<sup>~~without `level=0`, we can just add `, 0` like so `set_status_and_level(...code.., 0)`??~~</sup>
- ~~`extrude_length` => `purge_length` to match others~~ done in another **PR** 
- `dtostrf` => `p/w_float_t` -- ~~**M114.cpp**~~|**M48.cpp**

Update
---

follow up to "[BUG] DWIN ProUI: error: 'PROBE_OFFSET_ZMIN' was not declared in this scope #26505"  
in _ProUI_, created `_OFFSET_ZMIN/MAX` like in is in _CrealityUI_, made it so it = `PROBE_OFFSET_ZMIN/MAX` if `HAS_BED_PROBE`/defined, otherwise = `+/-20`

- optimize #define/#include in JyersUI and CrealityUI as well
- use `PROBE_SELECTED` to enabled `PROBE_OFFSET_ZMIN|MAX` (because --see below--`)
- simplified Print Progress (`_printtime`, `_remain_time`, `_percent_done`) in ProUI
<sup>may need to adjust layout position on the UI</sup>  
- - made use of `duration_t` -- `toDigital`, and `pcttostrpctrj`
- - benefits: less bytes, also adds Days to print time elapsed/remaining
- update **bedlevel_tools.cpp**, utilize unused functions
- add function to `zeroPoint` (reset single mesh x/y coordinate)
- update trammingwizard:  
  shows correct screen while inuse - rather than user being able to still navigate menu  
  exits the wizard - displays within tolerance - if values are 0.00 as fail-safe, otherwise gives false reading.
  show's Lower/Raise in green/red.
- move `dwinIconShow` to respected header
- add to ESDiag: filament sensor "PRESENT" (green) / "Runout Detected" (red)
- - rename `ES_REPORT` => `ES_REPORTS` to not be confused with previously defined in **endstops.cpp**
- remove/replace `dwinPopupConfirm` w/ `dwinPopupContinue`
- typo in **module/probe.cpp** `Probe::run_z_probe()`. "-float" to "- float"
- swap `const char *` with `PGM_P` (which is defined as `const char *`)
- Tweak `CASELIGHT_USES_BRIGHTNESS` in **ProUI**, separate function that sets it into "apply" and "live"

- fix tramming wizard, so now:
>When x = 0 and y = 0, p will be calculated as p = 0 + 2 * 0 = 0.  
>When x = 0 and y = 1, p will be calculated as p = 1 + 2 * 0 = 1.  
>When x = 1 and y = 0, p will be calculated as p = 0 + 2 * 1 = 2.  
>When x = 1 and y = 1, p will be calculated as p = 1 + 2 * 1 = 3.

>[!NOTE]
>~~*I'm not so sure about this one*~~
> - ~~in **pinsDebug_list.h** I added Maple support which have their SDIO defined as `BOARD_SDIO_*`~~
```diff
 #if PIN_EXISTS(SDIO_D0)
   REPORT_NAME_DIGITAL(__LINE__, SDIO_D0_PIN)
+#elif defined(BOARD_SDIO_D0)
+  REPORT_NAME_DIGITAL(__LINE__, BOARD_SDIO_D0)
 #endif
```

<br>

in **Marlin\src\inc\Conditionals_post.h**  
you will see
```y
#if !HAS_LEVELING || ANY(MESH_BED_LEVELING, AUTO_BED_LEVELING_UBL)
  #undef PROBE_MANUALLY
#endif
#if ANY(HAS_BED_PROBE, PROBE_MANUALLY, MESH_BED_LEVELING)
  #define PROBE_SELECTED 1
#endif
```


---
removed `HAS_ONESTEP_LEVELING`, replaced with `HAS_AUTOLEVEL` because  
(previously...)

```y
#if DISABLED(PROBE_MANUALLY) && ANY(AUTO_BED_LEVELING_BILINEAR, AUTO_BED_LEVELING_LINEAR, AUTO_BED_LEVELING_3POINT, AUTO_BED_LEVELING_UBL)
  #define HAS_ONESTEP_LEVELING 1
#endif
```
this code is the same as `HAS_AUTOLEVEL`, as you can see here:
```y
#if ANY(HAS_ABL_NOT_UBL, AUTO_BED_LEVELING_UBL)
  #define HAS_ABL_OR_UBL 1
  #if DISABLED(PROBE_MANUALLY)
    #define HAS_AUTOLEVEL 1
  #endif
#endif
```
<br>


<!--

Clearly describe the submitted changes with lots of details. Include images where helpful. Initial reviewers may not be familiar with the subject, so be as thorough as possible. You can use MarkDown syntax to improve readability with bullet lists, code blocks, and so on. PREVIEW and fix up formatting before submitting.

-->

### ~~Requirements~~

<!-- Does this PR require a specific board, LCD, etc.? -->

### ~~Benefits~~

<!-- What does this PR fix or improve? -->

### ~~Configurations~~

<!-- Attach Configurations ZIP and any other files needed to test this PR. -->

### Related Issues
if anyone wishes to take another look, these are what I suggest:  
see if **buildroot/tests/STM32F103RE_creality** needs anything  

and also...

in JyersUI/**dwin.cpp**  
`MAX_Z_OFFSET` seemed out of place without `MIN_Z_OFFSET`
basically these values can be used even without `HAS_Z_OFFSET_ITEM` (`BABYSTEPPING`), it would cause an error being undefined for some Configurations

```diff
-#if HAS_ZOFFSET_ITEM
-  #define MAX_Z_OFFSET 9.99
-  #if HAS_BED_PROBE
-    #define MIN_Z_OFFSET -9.99
-  #else
-    #define MIN_Z_OFFSET -1
-  #endif
-#endif

+#if HAS_ZOFFSET_ITEM && HAS_BED_PROBE
+  #define MAX_Z_OFFSET  20
+  #define MIN_Z_OFFSET -20
+#else
+  #define MAX_Z_OFFSET  3
+  #define MIN_Z_OFFSET -3
+#endif
```


<!-- Does this PR fix a bug or fulfill a Feature Request? Link related Issues here. -->

classicrocker883 committed:
general small fixes/tweaks
remove whitespace, add hyphen (-)
remove unused `INLINE_USART_IRQ`
optimize code

classicrocker883 committed:
Update Probe Offset dependencies
have `HAS_Z_OFFSET_ITEM` defined in one file
undef PROBE_OFFSET_ZMIN when ProUI is disabled

classicrocker883 committed:
edit Conditionals_post.h

thisiskeithb commented on docs/Cutter.md:
```diff
@@ -51,10 +51,10 @@ The following flow charts depict the flow control logic for spindle and laser op
+   │G-Code│  │  │Continuous│  │M3 receive │  │sets block │  │sync power ?│  │sync power ?│  │trap power ?│  │uses block │
+   │Send  ├──┼──┤Inline    ├──┤power from ├──┤power using├──┤process M3  ├──┤process fan ├──┤adjusts for ├──┤values to  │
+   └──────┘  │  │          │  │S-Value    │  │Gx S-Value │  │power inline│  │power inline│  │accel/decel │  │apply power│
              │  └──────────┘  └───────────┘  └───────────┘  └────────────┘  └────────────┘  └────────────┘  └───────────┘
```
This is broken now:

<img width="1082" alt="image" src="https://github.com/MarlinFirmware/Marlin/assets/13375512/7ef67dce-fc6e-4823-bfa3-8eb0f5d71953">

There's also a "gcode" in: https://github.com/MarlinFirmware/Marlin/blob/483b8dcc05c2f0ef4c83d8ec7f4f0065ec9d240f/docs/Cutter.md?plain=1#L7

Fixed in https://github.com/MarlinFirmware/Marlin/pull/26502/commits/87dbb8e20747d8c1df404371f6c70f0dd109ba2b

thisiskeithb committed:
Fix Cutter.md, more gcode => G-code

classicrocker883 commented:
> Sounds like these Ender-3 V2 UIs need to decouple from requiring/assuming a probe is installed instead of this workaround.

it uses that probe offset value regardless of one or not, using it for changing the Z-offset.

how about using a `TERN`?

classicrocker883 committed:
omit PROBE_OFFSET_ZMIN/MAX

classicrocker883 committed:
change to HAS_LEVELING - creality dwin.cpp

classicrocker883 committed:
replace HAS_ONESTEP_LEVELING => HAS_AUTOLEVEL
"#if DISABLED(PROBE_MANUALLY) && ANY(AUTO_BED_LEVELING_BILINEAR, AUTO_BED_LEVELING_LINEAR, AUTO_BED_LEVELING_3POINT, AUTO_BED_LEVELING_UBL)" is literally HAS_AUTOLEVEL

thinkyhead commented on Marlin/src/lcd/e3v2/proui/dwin.h:
```diff
@@ -234,7 +234,7 @@ void doCoolDown();
 #endif
-#if HAS_ONESTEP_LEVELING
+#if HAS_AUTOLEVEL
   void trammingwizard();
```
Note that tramming is not leveling. And "one step" leveling is any `G29` procedure that can be completed without human intervention. So the tramming wizard should not be associated with `HAS_AUTOLEVEL` or `HAS_ONESTEP_LEVELING`.

It may help to bring @mriscoc into the conversation about changes to ProUI since some of these ProUI flags may be modified to use existing configuration options that apply to other UIs.

classicrocker883 replied:
this is the "trammingwizard", it used the bed probe to check each corner.  
if one wanted to tram without a having a bed probe, the menu still gives that option. the wizard is just the auto version by checking level like it would for a mesh.

classicrocker883 replied:
![Screenshot 2023-12-08 040838](https://github.com/MarlinFirmware/Marlin/assets/18502096/7ade20eb-a764-4b6c-86d5-b7fb00826b2b)

thinkyhead replied:
The "tramming wizard" may involve the probe, but it is a feature independent of leveling, and it should be possible to disable it even if a probe exists and leveling is enabled. These things are separate in the configurations, and I have been encouraging the authors of UI code to respect these settings in our discussions. We should continue to encourage and develop in the direction of properly applying configuration settings rather than perpetuating the oversights of UI authors.

mriscoc replied:
> It may help to bring @mriscoc into the conversation about changes to ProUI since some of these ProUI flags may be modified to use existing configuration options that apply to other UIs.

I currently use:
```cpp
#if ENABLED(LCD_BED_TRAMMING)
  void drawTrammingMenu();
  #if HAS_TRAMMING_WIZARD
    void runTrammingWizard();
  #endif
#endif
```
That allows me to disable the wizard independently to save program memory.

thinkyhead commented on Marlin/src/inc/Conditionals_post.h:
```diff
@@ -2946,7 +2946,11 @@
  */
-#if ANY(MESH_BED_LEVELING, HAS_BED_PROBE)
+#if ANY(BABYSTEPPING, PROBE_SELECTED)
+  #define HAS_ZOFFSET_ITEM 1
```
Different UIs are going to have different criteria for whether or not they display a "Z Offset" item, so this flag should be left with each individual UI that currently uses it.

classicrocker883 replied:
that may be true, however the only UIs which have `HAS_ZOFFSET_ITEM` are exactly those three. that I am able to find.

thinkyhead replied:
Yep, and all those UIs are externally and independently developed, no guarantee they'll remain in sync. So these flags should be left with their respective UIs.

thinkyhead commented on Marlin/src/inc/Conditionals_post.h:
```diff
@@ -2946,7 +2946,11 @@
+
+#if HAS_ZOFFSET_ITEM
   #ifndef PROBE_OFFSET_ZMIN
     #define PROBE_OFFSET_ZMIN -20
```
These value range settings should only pertain to editing of the Probe Z Offset. Note that a UI may include both a baby-stepping item that has no effect on the probe offset, and a Probe Z Offset adjustment that doesn't baby-step. Non-probe-offset-adjusting baby-stepping is not limited to any range.

classicrocker883 replied:
the main concern is `DWIN_LCD_PROUI` uses `PROBE_OFFSET_ZMIN/MAX` whether it `HAS_MESH`, a probe --auto/mesh bed leveling, or none of the above. because it uses that offset value to adjust the Z-Offset (nozzle to bed relation).

I mean if this **PR** change isn't good, we could always just redefine it in **dwin_defines.h** into something else like  
```y
#ifndef PROBE_Z_OFFSET_ZMIN
  #define Z_OFFSET_ZMIN   20
#else
  #define Z_OFFSET_ZMIN  PROBE_Z_OFFSET_ZMIN
#endif
```
and use the new value in for the max/min of the function

thisiskeithb replied:
[As I mentioned above](https://github.com/MarlinFirmware/Marlin/pull/26502#issuecomment-1846055677), these UIs need to decouple from requiring/assuming a probe is installed so we don't end up with these kind of bug reports when someone doesn't have specific hardware or enable a specific feature:

- https://github.com/MarlinFirmware/Marlin/issues/26505

thinkyhead commented on Marlin/src/lcd/e3v2/creality/dwin.cpp:
```diff
@@ -1139,7 +1131,7 @@ void popupWindowHome(const bool parking/*=false*/) {
 }
 
-#if HAS_ONESTEP_LEVELING
+#if HAS_AUTOLEVEL
```
This includes UBL but unfortunately UBL is rarely a one step leveling process. It usually requires an auto leveling step followed by a manual leveling step. So some additional checking must be done to ensure that UBL is actually one-step (i.e., it can reach all mesh points with the probe) or additional UI must be developed to guide users through the following steps.

classicrocker883 replied:
so then it makes sense to be `HAS_AUTOLEVEL`?  
as far as I understand it, UBL doesnt *have* to be followed by manual leveling. the auto level option usually probes all the points, and any points not able to reach, it fills in the _unprobed_ values based on the surrounding _probed_ points, and in my experience with great accuracy.

for Bilinear or UBL, its usually the default, and one and done kind of thing. and if they wanted to check the points or do it manually they have that option as well.

thinkyhead replied:
UBL only fills in the un-probed values when commanded to do so by a call to `G29 P3`. I don't think this old Creality screen implementation does that.

thinkyhead committed:
Update Conditionals_post.h

thinkyhead committed:
serial tweaks

classicrocker883 committed:
Conditionals_post.h replace HAS_ZOFFSET_ITEM => PROBE_SELECTED

classicrocker883 committed:
Update Conditionals_post.h, include default-no probe/no mesh

if one had no Probe and no Mesh, they can still adjust their Z-Offset

classicrocker883 committed:
add default-no probe, no mesh to tests

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
simplify MPC/PID Plot, dwin.cpp

classicrocker883 committed:
add HAS_TRAMMING_WIZARD
swap to TERN_ in switch case

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
update dwin.cpp spacing

classicrocker883 committed:
update DWIN UI - spacing
removed COLOR_AQUA - unused, same as COLOR_CYAN

classicrocker883 committed:
update ProUI, remove unused, use Move/Probe methods

classicrocker883 committed:
revert #undef, change to _OFFSET_ZMIN|MAX
fix JyersUI - MAX|MIN_Z_OFFSET

classicrocker883 committed:
tweak spacing

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 commented:
I wanna say this **PR** is ready to go. please take another look through

classicrocker883 committed:
small change
change from OPTCODE to #if

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
fix error, duplicate case

classicrocker883 committed:
rename to G-Code, Marlin webui

classicrocker883 committed:
update digitpot spacing

classicrocker883 committed:
update more spacing

classicrocker883 committed:
add ZeroMesh in editmesh menu
ResetMesh if HAS_MESH

classicrocker883 committed:
add space to "extern"

classicrocker883 committed:
remove extra ";" from invaders.h

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
update pins_CREALITY_CR4NTXXC10.h w/HAS_DWIN_E3V2

classicrocker883 committed:
remove DWIN_CREALITY_LCD from MarlinCore.cpp, use ui.update() - relocate in dwin.cpp

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
revert Configuration_adv.h

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
fix CrealityUI error - ui.update/kill_screen
update tests capitilize "v2"
update so CrealityUI => `HAS_DISPLAY`, uses ui.update

classicrocker883 committed:
fix proui error

classicrocker883 committed:
fix other proui error

classicrocker883 committed:
transfer from Enable 2 mesh viewers as fixes/tweaks

classicrocker883 committed:
remove comment (unused kill_screen)

classicrocker883 committed:
add M48 pause, return if error

classicrocker883 committed:
removed because result = default

classicrocker883 committed:
no such ZHOME_BEFORE_LEVELING, removed

classicrocker883 committed:
fix for redef, rename SDIO_

classicrocker883 committed:
unused option for smooth encoder added
left commented out, not recommended unless user rather have it enabled

classicrocker883 committed:
update dwin.cpp - add HAS_FLOW/FEEDRATE_EDIT
rearrange layout

classicrocker883 committed:
remove space

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
fix for error, fontWidth/Height not used by these UI

classicrocker883 committed:
update print progress

classicrocker883 committed:
update print progress - simplify

classicrocker883 committed:
update ProgressElapsed

classicrocker883 committed:
update dwin.cpp, bedleveltools

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 commented:
ok I want to say this is ready. at least for another review.

classicrocker883 committed:
add to pinsDebug_list.h

classicrocker883 committed:
move dwinIconShow to dwin_lcd.h, replace dwinPopupConfirm w/ dwinPopupContinue, rename ICON_* functions

classicrocker883 committed:
change signed float to long

classicrocker883 committed:
fix for manualMeshSave

classicrocker883 committed:
move previous up

classicrocker883 committed:
fix convert error

classicrocker883 committed:
revert pins_trigorilla

classicrocker883 committed:
update ESDiag - is filament sensor?

thinkyhead commented on Marlin/src/gcode/lcd/M0_M1.cpp:
```diff
@@ -73,9 +73,9 @@ void GcodeSuite::M0_M1() {
   #elif ENABLED(DWIN_LCD_PROUI)
     if (parser.string_arg)
-      dwinPopupConfirm(ICON_BLTouch, parser.string_arg, GET_TEXT_F(MSG_USERWAIT));
+      dwinPopupContinue(ICON_BLTouch, F(parser.string_arg), GET_TEXT_F(MSG_USERWAIT));
```
C strings, `PROGMEM` strings, and `F` strings all need to be treated differently. If `dwinPopupContinue` wants to do something with a C string then there must a separate `dwinPopupContinue` method that takes a C string. If we wanted a method that accepts a `PROGMEM` string (`PGM_P`) it must be named `dwinPopupContinue_P` because `PROGMEM` strings cannot be distinguished from regular C strings. `PROGMEM` strings can be wrapped in `FPSTR()` to be passed to a method taking an `F` string. And `F` strings can be wrapped in `FTOP()` to be passed to a method taking a `PROGMEM` string.

thinkyhead commented on Marlin/src/lcd/e3v2/proui/dwin_popup.h:
```diff
@@ -69,11 +69,3 @@ void dwinShowPopup(const uint8_t icon, T amsg1=nullptr, U amsg2=nullptr, uint8_t
-  hmiSaveProcessID(ID_WaitResponse);
-  dwinDrawPopup(icon, amsg1, amsg2, BTN_Confirm);  // Button Confirm
-  dwinUpdateLCD();
-}
```
Templated functions like this can be used to do double duty for C strings and F strings (assuming `dwinDrawPopup` is also set up to do both C and F strings in a non-templated way (or does checking with `__typeof__` within the templated function).

thinkyhead commented on Marlin/src/pins/pinsDebug_list.h:
```diff
@@ -2300,21 +2300,36 @@
 #if PIN_EXISTS(SDIO_D0)
   REPORT_NAME_DIGITAL(__LINE__, SDIO_D0_PIN)
+#elif defined(BOARD_SDIO_D0)
+  REPORT_NAME_DIGITAL(__LINE__, BOARD_SDIO_D0)
```
Only pins that are actually assigned to a Marlin function need to be reported here. We don't need to report pins defined with a `BOARD_` prefix, as they are merely intended as defaults if not otherwise defined.

thinkyhead replied:
Currently the SDIO pins are defined by the HAL and then in some instances pins files define some `BOARD_SDIO_*` pins (without a `_PIN` suffix). These break the pattern that we had previously established, so it would be good to fix these up. The way it should work is:
- Define `BOARD_SDIO_*_PIN` pins in any pins files that want to override the HAL.
- Define `SDIO_*_PIN` as `BOARD_SDIO_*_PIN` when such a define exists,
- Otherwise, define `SDIO_*_PIN` based on the MCU.

I see that some of these are defined by `variants/*/board.h` while `pins_AQUILA_101.h` is the only pins file that does so. It's not entirely clear where in the build process `variants/*/board.h` is included, but I presume that it's ahead of all Marlin sources since it's part of the platform support. So, the common approach should work as described above for all the `SDIO_*_PIN` defines.

thinkyhead replied:
That said, caveats like this should not hold up this PR. Any items in this sweeping PR that are still under discussion can be moved to a separate PR as necessary.

classicrocker883 replied:
I thought I had since reverted this. so I'll go through this once over to see if there is anything left to do. otherwise if you can point anything out, or list it what needs to be changed lmk

thinkyhead commented on Marlin/src/lcd/e3v2/proui/endstop_diag.cpp:
```diff
@@ -57,11 +57,12 @@ void draw_es_label(FSTR_P const flabel=nullptr) {
-  is_hit ? DWINUI::drawString(RGB(31,31,16), F(STR_ENDSTOP_HIT)) : DWINUI::drawString(RGB(16,63,16), F(STR_ENDSTOP_OPEN));
+  dwinDrawRectangle(1, hmiData.colorPopupBg, LM, DWINUI::cursor.y, LM + 120, DWINUI::cursor.y + 20);
+  is_fil ? (is_hit ? DWINUI::drawString(RGB(16, 63, 16), F("PRESENT")) : DWINUI::drawString(RGB(31, 31, 16), F("Runout Detected"))) :
+  (is_hit ? DWINUI::drawString(RGB(31, 31, 16), F(STR_ENDSTOP_HIT)) : DWINUI::drawString(RGB(16, 63, 16), F(STR_ENDSTOP_OPEN)));
```
If these are for display on the LCD to common users they should use translatable strings with `GET_TEXT_F` rather than inline strings or `STR_*` strings that are intended for serial output.

classicrocker883 replied:
other than making up new ones, I'm looking through what is available, what about `MSG_YES` for hit, and `MSG_NO` for open?

as for "PRESENT" => `MSG_FILAMENT` and "Runout Detected" => `MSG_RUNOUT_SENSOR`, or it could be `MSG_FILAMENT_CHANGE_HEADER_LOAD` ("LOAD FILAMENT")

thinkyhead replied:
Making up new ones is the correct way to go for strings not yet defined. Prefer pre-existing ones where possible. I went ahead and added a commit to internationalize various strings. If you find any more that need translation as you go along, please add them.

classicrocker883 replied:
im going to leave the `STR_*` for now because a. I can't find a suitable `MSG_` replacement, and b. im sure "TRIGGERED" in big red letters is universally understood 😆.

thinkyhead commented on Marlin/src/lcd/e3v2/proui/dwin.cpp:
```diff
@@ -68,10 +69,6 @@
 
-#if DISABLED(PROBE_MANUALLY) && ANY(AUTO_BED_LEVELING_BILINEAR, AUTO_BED_LEVELING_LINEAR, AUTO_BED_LEVELING_3POINT)
-  #define HAS_ONESTEP_LEVELING 1
-#endif
```
We should keep this define. If you've modified the LCD code to call `G29 P1\nG29 P3` for UBL then UBL can be considered "one step" but the ABL options with `PROBE_MANUALLY` are definitely not one-step leveling procedures. `PROBE_MANUALLY` requires user interaction with the encoder to measure each point and proceed to the next point on click.

classicrocker883 replied:
I do not see anywhere in ProUI that `HAS_ONESTEP_LEVELING` is being utilized. its currently defined like this in two seperate files. even the current repo at mriscoc's ProUI doesnt have that and it has no use for `PROBE_MANUALLY`. when UBL or ABL is used, its automatically understood you have that option...to use it as a "one step" or not. just like Mesh Bed Leveling.  
which is also called Manual Mesh Bed Leveling right? since its manually probing the bed without an actual probe.

what is confusing is the `#define PROBE_MANUALLY` option in configuration.h:
>The "Manual Probe" provides a means to do "Auto" Bed Leveling without a probe.  


it sounds like if you have `MESH_BED_LEVELING`, you would want to also enable this. yet in **inc/Conditionals_LCD.h**:
```y
#if !HAS_LEVELING || ANY(MESH_BED_LEVELING, AUTO_BED_LEVELING_UBL)
  #undef PROBE_MANUALLY
```

but besides that, a user having ABL or UBL should always have the option to "Probe manually" regardless.

that is how the menu is currently, except for maybe JyersUI. that is a bit confusing to have UBL, but also being able to not having any bed probe defined?

in summary, I think this should be left specifically for `MESH_BED_LEVELING`; if you dont have a probe, that should be your only option for mesh leveling.  
and in regards to Auto bed leveling w/probe, it should always give you that "probe manually" option anyway in the menu/gcode. in otherwords, if it `HAS_LEVELING` the "probing manually" is always an option. 

I'm just saying this is how it is set up with ProUI right now
- Probe?
UBL, or ABL
- No Probe?
Mesh Bed Leveling, or None

I'm also able to compile using `MESH_BED_LEVELING` while having a probe enabled. is that supposed to be?

It seems that things aren't what they used to be, or meant to be anymore. or maybe I have it all wrong. like was `AUTO_BED_LEVELING_UBL` supposed to be used without a bed probe like it can be in JyersUI? if so, why not just use `MESH_BED_LEVELING` instead?

on a side note, even Mriscoc is phasing out using Bilinear auto bed leveling in favor of UBL. I know their code is different, but on the surface its the same thing. *you probe, and you get a mesh.*  
plus apparently there seems to be some issues with saving the mesh. it doesn't load properly, and you have to manually load settings otherwise you get an invalid mesh.

mriscoc replied:
> plus apparently there seems to be some issues with saving the mesh. it doesn't load properly, and you have to manually load settings otherwise you get an invalid mesh.

Currently the paradigm is different, due to the addition of AML 2.0 support, now all mesh parameters are saved in the UBL slot, so you can have slots with different mesh sizes and mesh densities. You can know how it is working now and the new way of use UBL in the Professional firmware here: https://github.com/mriscoc/Ender3V2S1/wiki/Mesh-Leveling-Menu#ubl

thinkyhead replied:
I picked this line at random to comment on the define, but the general idea is to keep using this define in code that currently uses it, wherever that is.

thinkyhead replied:
> I'm also able to compile using `MESH_BED_LEVELING` while having a probe enabled. is that supposed to be?

Yes, that option is specifically for a machine that has no probe. You can also use `PROBE_MANUALLY` with any ABL option (other than UBL) for a machine that has no probe. In both cases, you send a `G29` command to get each point, and keep doing this in a loop until the procedure is completed.

See `menu_bed_leveling.cpp` within the `ANY(PROBE_MANUALLY, MESH_BED_LEVELING)` block to see how this is implemented from the UI.

thinkyhead replied:
> due to the addition of AML 2.0 support

Good that you have a wiki to cover the topic. My aim with ABL Bilinear was to make a leveling system that can be done all in one go based on a mesh that only covers the area of the bed that the probe can reach. It actually is "adaptive" in that you can set the boundaries of the mesh that you want to probe, but it always probes the same number of points.

We have a PR pending to make all leveling systems runtime-configurable in both the boundaries and the number of points in X and Y, but of course it's slow going because every LCD implementation wants to do things its own way. I cannot stress enough how important it is to work with the main Marlin project to standardize these things in the core and not to keep "going it alone." So anything you can do towards that effort would be much appreciated. I would do all of this myself, but daily tasks take up most of the time I would otherwise spend making awesome new code.

classicrocker883 replied:
> I would do all of this myself, but daily tasks take up most of the time I would otherwise spend making awesome new code.

this quote 😂  
oh good, I'm not alone

thinkyhead committed:
misc. adjustments

thinkyhead committed:
extra ui.*

thinkyhead committed:
Fix dwinPopupContinue

classicrocker883 committed:
dwinPopupContinue is undefined without #include dwin_popup.h

classicrocker883 committed:
remove unneeded highlightY

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

thinkyhead committed:
🌐 Internationalize

thinkyhead committed:
prettify

classicrocker883 committed:
update spanish lang

classicrocker883 committed:
missing _ typo error

classicrocker883 committed:
revert pinsDebug_list

classicrocker883 committed:
remove bool from readEEPROM

classicrocker883 committed:
4 arg dwinIconShow unused, rename dacaiIconShow

thinkyhead commented:
Thanks for adding more languages. Take care to correct any strings that are over 20 characters, as these must be shortened to 20 characters or fewer. If a longer string is preferred for wider strings, put the wider strings into the LanguageWide namespace.

thinkyhead commented on Marlin/src/lcd/e3v2/proui/dwin.cpp:
```diff
@@ -2052,14 +2040,14 @@ void gotoConfirmToPrint() {
-    const bool success = settings.load();
     dwinRedrawScreen();
-    DONE_BUZZ(success);
+    DONE_BUZZ(settings.load());
```
`settings.load` should come ahead of directly redrawing the screen, because otherwise any screen that displays or uses settings will be displaying / using old settings.

thinkyhead commented on Marlin/src/lcd/e3v2/proui/dwin.cpp:
```diff
@@ -2069,7 +2057,10 @@ void gotoConfirmToPrint() {
   #if HAS_MESH
-    void saveMesh() { TERN(AUTO_BED_LEVELING_UBL, ublMeshSave(), writeEEPROM()); }
+    void saveMesh() {
+      TERN_(MESH_BED_LEVELING, manualMeshSave();)
```
Keep semicolons at the end of lines so that linters won't bark at us. If there is some situation where the presence of a semicolon would be bad (e.g., inside of a `switch` block) then use `OPTCODE` instead of `TERN_` and the semicolon will be added automagically.

thinkyhead committed:
fix some code wrappers

thinkyhead commented on Marlin/src/lcd/e3v2/proui/dwin.cpp:
```diff
@@ -2555,7 +2559,7 @@ void setFlow() { setPIntOnClick(FLOW_EDIT_MIN, FLOW_EDIT_MAX, []{ planner.refres
   void manualMeshSave() {
     LCD_MESSAGE(MSG_UBL_STORAGE_MESH_MENU);
-    queue.inject(F("M211S1\nM500"));
+    queue.inject(F("M211S1"));
```
This doesn't pertain to UBL. See line 2061.
```cpp
TERN_(MESH_BED_LEVELING, manualMeshSave());
```

thinkyhead replied:
Also, `M211S1` enables software endstops. I'm not sure how that is pertinent here.

classicrocker883 replied:
well, `manualMeshStart` (line 2537) disables them inorder for it to do its bed leveling. 

>This doesn't pertain to UBL. See line 2061.

sorry, I don't follow...

classicrocker883 committed:
wrappers cont.

thinkyhead committed:
Merge branch 'bugfix-2.1.x' into pr/26502

thinkyhead committed:
Merge branch 'bugfix-2.1.x' into pr/26502

thinkyhead committed:
move test

thinkyhead committed:
…more…

thinkyhead committed:
if HAS_DWIN_E3V2 => HAS_DISPLAY then…

thinkyhead committed:
…and…

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
move up settings.load

classicrocker883 reviewed (commented):
> Thanks for adding more languages. Take care to correct any strings that are over 20 characters, as these must be shortened to 20 characters or fewer. If a longer string is preferred for wider screens, put the wider strings into the LanguageWide namespace. 

I thought I would do at least a couple, but seeing it is a bit tedious for even one; going back and forth with google translate, and another, trying to get the words right... and since I only know one real language, I figure I'll leave the rest to someone else, or maybe if I get bored enough.

classicrocker883 committed:
update runout string

classicrocker883 committed:
revert change to popup

classicrocker883 committed:
update gcode_preview - fix for restarting after multiple clicks

classicrocker883 committed:
fix typo

classicrocker883 committed:
initialize msg array

classicrocker883 committed:
shortened some espanol
moved MSG_BLTOUCH_MODE_CHANGE to Wide
Note: some translations are 21 characters... not much I can do about it, so hope that might be ok

classicrocker883 committed:
fix tramwiz, so now...
When x = 0 and y = 0, p will be calculated as p = 0 + 2 * 0 = 0.
When x = 0 and y = 1, p will be calculated as p = 1 + 2 * 0 = 1.
When x = 1 and y = 0, p will be calculated as p = 0 + 2 * 1 = 2.
When x = 1 and y = 1, p will be calculated as p = 1 + 2 * 1 = 3.

classicrocker883 committed:
fix (-) float

classicrocker883 committed:
toggleCheckboxLine -- ^= true

classicrocker883 committed:
null msg, meshviewer.cpp

classicrocker883 committed:
OPTARG to update_software_endstops

classicrocker883 committed:
tweak trammingwizard

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
update printtime

classicrocker883 committed:
add millis_t to duration_t.h

classicrocker883 committed:
swap const char * w/ PGM_P - ignore previous commits of same

classicrocker883 committed:
revert PGM_P swap on gcode_preview

classicrocker883 committed:
tweak caselightbrightness

classicrocker883 committed:
revert duration_t.h for other PR

classicrocker883 committed:
change to update every second

classicrocker883 committed:
update ubl_g29.cpp comment spacing

classicrocker883 committed:
captialize letter

classicrocker883 committed:
update comments+spacing, ubl files

classicrocker883 committed:
update ubl_G29

classicrocker883 committed:
update ubl_G29

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
remove HAS_STATUS_MESSAGE w/HAS_DISPLAY, update DWIN, use ALL(...)

classicrocker883 committed:
add kill_screen to CrealityUI, group MarlinUI functions in JyersUI

classicrocker883 committed:
add dwinPopupConfirm/Continue, update Creality/JyersUI dwinDrawString

classicrocker883 committed:
#include in dwin.cpp

classicrocker883 committed:
Merge branch 'bugfix-2.1.x' of https://github.com/MarlinFirmware/Marlin into bugfix-2.1.x-Dec

classicrocker883 committed:
remove characters causing issue with make format-pins

classicrocker883 committed:
revert pin *.h

classicrocker883 committed:
Update jyersui dwin.cpp - manualValueUpdate

classicrocker883 committed:
Update proui bedlevel_tools.cpp - manualValueUpdate

classicrocker883 committed:
update proui spacing/comments

classicrocker883 committed:
Update jyers ui dwin.cpp

classicrocker883 committed:
Update dwin_popup.h spacing

thinkyhead committed:
Merge branch 'bugfix-2.1.x' into pr/26502

thinkyhead commented:
It is not possible to review all these changes and accept them all in one PR. You should cherry-pick parts of this large PR into smaller pieces with changes limited to specific areas of the code, or only addressing specific things, and submit those separately as smaller PRs.
